File: src/main/java/net/atos/zac/aanvraag/model/InboxProductaanvraagListParameters.java
/*
 * SPDX-FileCopyrightText: 2022 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.zac.aanvraag.model;

import net.atos.zac.shared.model.ListParameters;
import net.atos.zac.zoeken.model.DatumRange;


public class InboxProductaanvraagListParameters extends ListParameters {

    private DatumRange ontvangstdatum;

    private String type;

    private String initiatorID;


    public InboxProductaanvraagListParameters() {
        super();
    }

    public DatumRange getOntvangstdatum() {
        return ontvangstdatum;
    }

    public void setOntvangstdatum(final DatumRange ontvangstdatum) {
        this.ontvangstdatum = ontvangstdatum;
    }

    public String getType() {
        return type;
    }

    public void setType(final String type) {
        this.type = type;
    }

    public String getInitiatorID() {
        return initiatorID;
    }

    public void setInitiatorID(final String initiatorID) {
        this.initiatorID = initiatorID;
    }
}

File: src/main/app/src/app/productaanvragen/model/inbox-productaanvraag-list-parameters.ts
/*
 * SPDX-FileCopyrightText: 2022 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

import { SortDirection } from "@angular/material/sort";
import { ListParameters } from "../../shared/model/list-parameters";
import { DatumRange } from "../../zoeken/model/datum-range";
import { ZoekFilters } from "../../gebruikersvoorkeuren/zoekopdracht/zoekopdracht.component";

export class InboxProductaanvraagListParameters
  extends ListParameters
  implements ZoekFilters
{
  readonly filtersType = "InboxProductaanvraagListParameters";
  type: string;
  ontvangstdatum = new DatumRange();
  initiatorID: string;

  constructor(sort: string, order: SortDirection) {
    super(sort, order);
  }

  static heeftActieveFilters(zoekFilters: any): boolean {
    return (
      zoekFilters.type != null ||
      zoekFilters.ontvangstdatum?.van != null ||
      zoekFilters.ontvangstdatum?.tot != null ||
      zoekFilters.initiatorID != null
    );
  }
}
